### PR TITLE
CentOS 8: Install python3 version of aws-cfn-bootstrap

### PR DESCRIPTION
**Pull request by rexcsn**
* Install python3 version of aws-cfn-bootstrap scripts to support CentOS8

Signed-off-by: Rex <shuningc@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: amis/build_ami.sh [hunk 1]
```diff
@@ -184,8 +184,8 @@ check_options() {
     esac
 
     # Ensure the specified architecture-OS combination is valid
-    if [ "${_arch}" == "arm64" ] && [[ "${_os}" =~ ^centos[0-9]+ ]]; then
-      echo "Currently there are no CentOS arm64 AMIs available."
+    if [ "${_arch}" == "arm64" ] && [[ "${_os}" != "centos8" ]]; then
```
Comment (enrico-usai) [inline amis/build_ami.sh:187]: I'd use `[[ "${_os}" =~ ^centos[6-7]$ ]]`

**Review by enrico-usai — CHANGES_REQUESTED** [review]
As a general comment let's use a more descriptive title for the PR (including the Cento8 word).

I'd also add more details in the commit message and in th PR description about the changes we're doing, expecially about the reason why we're upgrading to aws-cfn-bootstrap-py3.

It will be useful whether if squashing the commits related to this feature into a single one or if keeping the commit as is.

File: amis/build_ami.sh (hunk 1 above)
Comment (rexcsn) [reply amis/build_ami.sh]: Done

File: amis/packer_centos8.json [hunk 2]
```diff
@@ -280,9 +280,9 @@
         "region=\"{{user `region`}}\"",
         "bucket=\"s3.amazonaws.com\"",
         "[[ ${region} =~ ^cn- ]] && bucket=\"s3.cn-north-1.amazonaws.com.cn/cn-north-1-aws-parallelcluster\"",
-        "curl --retry 3 -L -o /tmp/aws-cfn-bootstrap-latest.tar.gz https://${bucket}/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz",
-        "which pip2",
-        "if [ $? -eq 0 ]; then sudo pip2 install /tmp/aws-cfn-bootstrap-latest.tar.gz; else sudo pip install /tmp/aws-cfn-bootstrap-latest.tar.gz; fi"
+        "curl --retry 3 -L -o /tmp/aws-cfn-bootstrap-py3-latest.tar.gz https://${bucket}/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz",
+        "sudo dnf install -y python3-pip",
```
Comment (enrico-usai) [inline amis/packer_centos8.json:284]: should we also install pip3?

**Comment by enrico-usai** [comment]
Could you please change the commit message and the PR description to be more descriptive? 
E.g. "Install python3 version of aws-cfn-bootstrap scripts to support CentOS8"

File: amis/packer_centos8.json (hunk 2 above)
Comment (rexcsn) [reply amis/packer_centos8.json:284]: `python3-pip` is pip3. We are only installing pip here

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**